scripts/sync_bugzilla_to_github.py

```diff
@@ -0,0 +1,354 @@
+                    if DRY_RUN:
+                        issue = {}
+                    else:
+                        issue = self._repo.create_issue(**issue_info)
+                    self.mirror_issues[bugid] = issue
+                    return True
+        else:
+            changed_fields = [
+                field for field in issue_info if issue_info[field] != getattr(issue, field)]
+            if changed_fields:
+                # Note that this will close issues that have not open in bugzilla.
+                log('Updating mirror issue #{} for bz{id} (changed: {})',
+                    issue.number, changed_fields, **bug_info)
+                # Weird API thing where `issue.edit` accepts strings rather than label refs...
+                issue_info['labels'] = [l.name for l in issue_info['labels']]
+                # Explain why we are closing this issue.
+                if not DRY_RUN:
+                    if not bug_info['is_open'] and 'state' in changed_fields and 'resolution' in bug_info:
+                        issue.create_comment(SYNCED_ISSUE_CLOSE_COMMENT.format(resolution=bug_info['resolution']))
+                    issue.edit(**issue_info)
+                return True
+            else:
+                if VERBOSE_DEBUG:
+                    log('No change for issue #{}', issue.number)
+        return False
+
+    def _format_issue_info(self, bug_info, issue):
+        issue_info = {
+            'state': 'open' if bug_info['is_open'] else 'closed'
+        }
+        if 'summary' in bug_info:
+            issue_info['title'] = bug_info['summary']
+        else:
+            issue_info['title'] = 'Confidential Bugzilla issue'
+        if 'comment0' in bug_info:
+            issue_info['body'] = bug_info['comment0']
+        else:
+            issue_info['body'] = 'No description is available for this confidential bugzilla issue.'
+
+        if issue is None:
+            issue_info['labels'] = [self._label]
+        else:
+            issue_info['labels'] = issue.labels
+            if self._label not in issue.labels:
+                issue_info['labels'].append(self._label)
+
+        # Ensure we include a link to the bugzilla bug for reference.
+        issue_info['body'] += SYNCED_ISSUE_TEXT.format(**bug_info)
+
+        # Preserve any Jira sync lines in the issue body.
+        if issue is not None:
+            for ln in issue.body.split("\n"):
+                if ln.startswith(JIRA_ISSUE_MARKER):
+                    issue_info['body'] += '\n' + ln
+            # Jira seems to sometimes add a trailing newline, try to match it to avoid spurious updates.
+            if issue.body.endswith('\n') and not issue_info['body'].endswith('\n'):
+                issue_info['body'] += '\n'
+
+        return issue_info
+
+
+def sync_bugzilla_to_github():
+    # Find the sets of bugs in bugzilla that we want to mirror.
+    log('Finding relevant bugs in bugzilla...')
+    bugs = BugSet(os.environ.get('BZ_API_KEY'))
```

Comment:
@rfk were these set in Github secrets?

Reply:
The `GITHUB_TOKEN` is configured somehow in out CircleCI config, although I don't remember the details.

The `BZ_API_KEY`, I don't think actually use in CI, because I don't about the security implications of a bugzilla access key leaking into this infrastructure. It might be fine if appropriately limited in powers, but I certainly wouldn't put one of my own bugzilla API keys into CI config here.

(I would occasionally run this script by hand with `BZ_API_KEY` set in my local environment in order to make sure we weren't missing any confidential issues, although I haven't been in the habit of doing that lately).